mbuild/compound.py

```diff
@@ -69,53 +69,62 @@ def load(filename, relative_to_module=None, compound=None, coords_only=False,
     compound : mb.Compound
 
     """
+
+    if compound is None:
+        compound = Compound()
+
+    # First check if we are loading from an existing parmed or trajectory structure
+    type_dict = {pmd.Structure:compound.from_parmed,md.Trajectory:compound.from_trajectory}
+    if isinstance(filename_or_topology, Compound):
+        return filename_or_topology
+    elif isinstance(filename_or_topology, tuple(type_dict)):
+        type_dict[type(filename_or_topology)](filename_or_topology,coords_only=coords_only, **kwargs)
```

Comment:
I think the downside of checking for the type and key being equal is that it doesn't catch cases in which the type inherits from one, i.e. some `ParmEd` data structures inherit from `pmd.Structure:compound`, whereas `isinstance()` would catch it

Reply:
Thanks! I will try to fix the logic used here.